### ISSUE TITLE
Char should not be trivially valid

### ISSUE THREAD

**Issue by lehins** [NorfairKing/validity#78]
Currently in `validity` `Char` is trivially valid. But if you check the [`chr`](https://downloads.haskell.org/~ghc/8.8.2/docs/html/libraries/base-4.13.0.0/src/GHC-Char.html#chr) function it is expected to be:
```haskel
| isTrue# (int2Word# i# `leWord#` 0x10FFFF##) = C# (chr# i#)
```
While when read from memory only 31 bits are expected: [`readWideCharOffAddr#`](https://downloads.haskell.org/~ghc/8.8.2/docs/html/libraries/base-4.13.0.0/GHC-Exts.html#v:readWideCharOffAddr-35-)

In other words I can construct an invalid `Char#` that can mess things up:
```Haskell
λ> C# (chr# 2147483647#)
'\2147483647'
λ> chr 2147483647
*** Exception: Prelude.chr: bad argument: 2147483647
```
In fact it is possible to construct even larger 64bit `Char`s that are ven negative :slightly_smiling_face:
```haskell
λ> C# (chr# 9223372036854775807#)
'\9223372036854775807'
λ> C# (chr# -9223372036854775808#)
'\-9223372036854775808'
```

**Comment by NorfairKing** [comment]
It looks like the maxBound is quite small:
```
ord (maxBound :: Char)
1114111
```
Certainly smaller than 31 bits:
```
logBase 2 $ fromIntegral $ ord (maxBound :: Char)
20.087461546321563
```

**Comment by NorfairKing** [comment]
@lehins  I opened https://github.com/NorfairKing/validity/pull/79/files
Could you have a look?

**Comment by NorfairKing** [comment]
This has been implemented in the layest release
